code/script.js

```diff
@@ -1,47 +1,252 @@
+  showMessage(`Hello there! Tomorrow's options are the following:`, 'bot')
+
+inputWrapper.innerHTML=
+`<button id="tulipsBtn">Bouquet of Tulips 15€</button>
+<button id="rosesBtn">Bouquet of Roses 25€</button>
+<button id="liliesBtn">Bouquet of Lilies 20€</button>`
+
 // Set up your eventlisteners here
+document.getElementById('tulipsBtn').addEventListener('click', () => {
+  flowers="Bouquet of Tulips 15€";
+  nextQuestion('Bouquet of Tulips 15€')
+})
+document.getElementById('rosesBtn').addEventListener('click', () => {
+  flowers="Bouquet of Roses 25€";
+  nextQuestion('Bouquet of Roses 25€')
+})
+document.getElementById('liliesBtn').addEventListener('click', () => {
+  flowers="Bouquet of Lilies 20€";
+  nextQuestion('Bouquet of Lilies 20€')
+})
+}
+
+// Question number 2
+const deliveryAddress = ()=> {
+  questionNumber++
+  botReply(`What is the delivery address?`)
+  
+  inputWrapper.innerHTML=
+  `<form id="formAddress">
+    <input id="inputAddress" type="text" />
+    <button class="send-btn" type="submit" id="send">Send</button>
+  </form>
+  `
+  document.getElementById('formAddress').addEventListener('submit', (event) => {
+    event.preventDefault()
+    address = inputAddress.value
+    nextQuestion(inputAddress.value)
+  })
+}
+
+// Question number 3
+const deliveryPerson = ()=> {
+  questionNumber++ //this is the same than writing questionNumber = questionNumber + 1
+  botReply(`Are the flowers for yourself or someone else?`)
+
+  inputWrapper.innerHTML=
+`<button id="myselfBtn">Myself</button>
+<button id="someoneElseBtn">Someone else</button>`
+
+//if the flowers are for myself bot does not ask the card questions and goes straight to question number 5
+document.getElementById('myselfBtn').addEventListener('click', () => {
+  questionNumber=5;nextQuestion('myself')
+})
+
+document.getElementById('someoneElseBtn').addEventListener('click', () =>{
+  nextQuestion('someone else')
+})
+} 
+
+//Question 4
+const card = () => {
+  questionNumber++
+  
+  botReply(`Would you like to add a greeting card with your flowers?`)
+  
+  inputWrapper.innerHTML=
+  `<button id="cardYesBtn">Yes</button>
+  <button id="cardNoBtn">No</button>`
+  
+  document.getElementById('cardYesBtn').addEventListener('click', () => {
+  cardYes="Yes"
+  
+  nextQuestion('Yes')
+  })
+
+  document.getElementById('cardNoBtn').addEventListener('click', () => {
+    questionNumber=5;
+    nextQuestion('No')
+  }) 
+}
+
+// Question number 5
+const cardGreeting = () => {
+  questionNumber++
+  
+  botReply (`Type your greeting here, for example ''Happy Birthday Lisa, Love Tom''`)
+  
+  inputWrapper.innerHTML=
+  `<form id="formCardGreeting">
+    <input id="inputField" type="text"/>
+    <button id="sendBtn" class="send-btn" type="submit">Send</button>
+  </form>`
+  
+  document.getElementById('formCardGreeting').addEventListener('submit', (event) => {
+    event.preventDefault()
+    cardText= inputField.value
+    nextQuestion(inputField.value)
+  })
+}
```

Comment:
We are thinking if you may remove questionNumber = 5, and then later clarify 2 cases (Yes and No) it in question number 5 (cardGreeting function).

if (message === "yes").
{ do something;
nextQuestion++ }
else { nextQuestion++ }

We thought that if you can test it and see if it works. As it may be easier to keep track of the question flow